salt/modules/influx.py

```diff
@@ -262,7 +263,17 @@ def user_exists(
     users = user_list(database, user, password, host, port)
     if not isinstance(users, list):
         return False
-    return name in [u['name'] for u in users]
+
+    for user in users:
+        # the dict key could be different depending on influxdb version
+        username = user.get('user', user.get('name'))
+        if username:
+            if username == name:
+                return True
+        else:
+            log.warning('Could not find username in user: %s', user)
```

Comment:
Is it intended to warn on every single user in the users list? This seems a little noisy. Would it make more sense to move this out of the loop and warn at the end if we haven't returned already?